shapely/coords.py

```diff
@@ -116,7 +116,10 @@ def __getitem__(self, key):
     def _ctypes(self):
         self._update()
         has_z = self._ndim == 3
-        n = self._ndim
+        n = self._ndim or 0
+        if n == 0:
```

Comment:
This is the same edit as with `shapely/speedups/_speedups.pyx`. It keeps and tests `n` as an integer, rather than testing `if n is None` (and yes, `self._ndim is None` for empty geometries).